src/server/config/config.js

```diff
@@ -163,8 +187,23 @@ module.exports = class Config {
 
   getSchema() {
     if (!this[schema]) {
-      let objKeys = zipObject([...this[schemaKeys]]);
-      this[schema] = Joi.object().keys(objKeys).default();
+      this[schema] = (function convertToSchema(children) {
+        let objKeys = zipObject([...children]);
```

Comment:
After tinkering with this some more, it seems that `Joi.object().keys().default()` will not validate the schema as we expect, but simply passing an empty array to keys, ala `Joi.object().keys([]).default()` will.